components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +246,180 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
         context = mock(PrivilegedCarbonContext.class);
         when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(context);
 
-        RegistryService registry = mock(RegistryService.class);
+        registry = mock(RegistryService.class);
         mockStatic(OAuth2ServiceComponentHolder.class);
         when(OAuth2ServiceComponentHolder.getRegistryService()).thenReturn(registry);
 
-        UserRegistry userRegistry = mock(UserRegistry.class);
+        userRegistry = mock(UserRegistry.class);
         when(registry.getConfigSystemRegistry(SAMPLE_TENANT_ID)).thenReturn(userRegistry);
 
-        Resource resource = mock(Resource.class);
+        resource = mock(Resource.class);
         when(userRegistry.get(OAuthConstants.SCOPE_RESOURCE_PATH)).thenReturn(resource);
 
         samlAssertionClaimsCallback.handleCustomClaims(jwtClaimsSet, oAuthAuthzReqMessageContext);
-        assertTrue(jwtClaimsSet.getAllClaims().isEmpty());
+        assertEquals(jwtClaimsSet.getAllClaims().size(), 8,
+                "Claims are not successfully set.");
+
+    }
+
+    @Test
+    public void testCustomClaimForOAuthTokenReqMessageContextWithNullAssertionSubject() throws Exception {
+        jwtClaimsSet = new JWTClaimsSet();
+        requestMsgCtx = mock(OAuthTokenReqMessageContext.class);
+
+        mockSubject = mock(Subject.class);
+        assertion = mock(Assertion.class);
+        nameID = mock(NameID.class);
+        when(assertion.getSubject()).thenReturn(mockSubject);
+        when(mockSubject.getNameID()).thenReturn(nameID);
+        when(nameID.getValue()).thenReturn(" C=US, O=NCSA-TEST, OU=User, CN=trscavo@uiuc.edu");
+
+        AttributeStatement statement = mock(AttributeStatement.class);
+        List<AttributeStatement> attributeStatementList = null;
+        when(assertion.getAttributeStatements()).thenReturn(attributeStatementList);
+
+        List<Attribute> attributesList = new ArrayList<>();
+        Attribute attribute = mock(Attribute.class);
```

Comment:
Why do we need to mock this ? Can't we use an instance of this ?

Reply:
We can not build an Attribute object because this does not have a setter to set attributes values.
Therefore inorder to return a List<XMLObject> form attribute, had to mock this attribute.getAttributeValues()